charts/redpanda/post_install_upgrade_job.go

```diff
@@ -0,0 +1,200 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=post-install-upgrade-job.go.tpl
+package redpanda
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+)
+
+func PostInstallUpgradeJob(dot *helmette.Dot) *batchv1.Job {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if !values.PostInstallJob.Enabled {
+		return nil
+	}
+
+	job := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-configuration", Fullname(dot)),
+			Namespace: dot.Release.Namespace,
+			Labels: helmette.Merge(
+				FullLabels(dot),
+				helmette.Default(map[string]string{}, values.PostInstallJob.Labels),
+			),
+			Annotations: helmette.Merge(
+				// This is what defines this resource as a hook. Without this line, the
+				// job is considered part of the release.
+				map[string]string{
+					"helm.sh/hook":               "post-install,post-upgrade",
+					"helm.sh/hook-delete-policy": "before-hook-creation",
+					"helm.sh/hook-weight":        "-5",
+				},
+				helmette.Default(map[string]string{}, values.PostInstallJob.Annotations),
+			),
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: fmt.Sprintf("%s-post-", dot.Release.Name),
+					Labels: helmette.Merge(
+						map[string]string{
+							"app.kubernetes.io/name":      Name(dot),
+							"app.kubernetes.io/instance":  dot.Release.Name,
+							"app.kubernetes.io/component": fmt.Sprintf("%s-post-install", Name(dot)[:50]),
```

Comment:
Fun fact, printf has builtin support for truncating `%[50]s-post-install`. Either way works for me.

Reply:
ah-ha, I'll fix that up.